core/textinput/src/textinput/Editor.cpp

```diff
@@ -116,11 +123,23 @@ namespace textinput {
       if (startAt == (size_t) -1) {
         startAt = 0;
       }
-      for (size_t i = startAt, n = Hist->GetSize(); i < n; ++i) {
-        if (Hist->GetLine(i).find(fSearch) != std::string::npos) {
-          NewHistEntry = i;
-          break;
-        }
+      if (fMode == kHistRevSearchMode)
+      {
+         for (size_t i = startAt, n = Hist->GetSize(); i < n; ++i) {
+           if (Hist->GetLine(i).find(fSearch) != std::string::npos) {
+             NewHistEntry = i;
+             break;
+           }
+         }
+      }
+      else
+      {
+         for (long i = startAt; i >= 0 ; --i) {
+            if (Hist->GetLine(i).find(fSearch) != std::string::npos) {
+               NewHistEntry = i;
+               break;
+            }
+         }
```

Comment:
Same here.  I have the feeling that these two loops can be merged (being the iteration order the only thing that changes).  :slightly_smiling_face: